src/PolyhedralGeometry/Polyhedron/standard_constructions.jl

```diff
@@ -949,6 +949,118 @@ julia> volume(p)
+
+# Examples
+```jldoctest
+julia> lambda = Partition([3,1,1])
+[3, 1, 1]
+
+julia> w0 = @perm (1,3,2)
+(1,3,2)
+
+julia> dc = demazure_character(lambda,w0)
+x1^3*x2*x3 + x1^2*x2^2*x3 + x1*x2^3*x3
+```
+"""
+function demazure_character(lambda::AbstractVector, sigma::PermGroupElem)
+       genGT = gelfand_tsetlin_polytope(lambda,sigma)
+       R = polynomial_ring(ZZ, length(lambda))
+       x = R[2]
+       s = R[1](0)
+       for P in lattice_points(genGT)
+          w=[sum([P[binomial(n+1,2)-binomial(i+1,2)+j] for j in 1:i]) for i in 1:length(lambda)]
+          s=s+x[1]^w[1]*prod([x[i]^(w[i]-w[i-1]) for i in 2:n])
+       end
+       return s
+end
+
+@doc Markdown.doc"""
+
+    gelfand_tsetlin_polytope(lambda::AbstractVector, w::PermGroupElem)
+
+Construct the generalized Gelfand Tsetlin polytope indexed by a weakly decreasing vector `lambda`
+and a permutation w.
+
+```jldoctest
+julia> P = gelfand_tsetlin_polytope([5,3,2],@perm (1,3,2))
+Polyhedron in ambient dimension 6
+```
+"""
+function gelfand_tsetlin_polytope(lambda::AbstractVector,w::PermGroupElem)
+   GT = gelfand_tsetlin_polytope(lambda)
+
+
+   #The i-th inversion set (when i=1 this is just inversions) as
+      #defined in Postnikov/Stanley '09
+   function inversions(σ; i=1)
+      n = σ.parent.deg
+      filter(x->σ(i)>σ(x),i:n)
+   end
+
+   #The code of a permutation as defined in Postnikov/Stanley '09
+   function code(σ)
+      [length(inversions(σ;i=i)) for i in 1:σ.parent.deg]
+   end
+
+   #The flag of a permutation as defined in Postnikov/Stanley '09
+   function flag(σ)
+      n=σ.parent.deg
+      w₀ = perm(vcat([n],collect(1:n-1)))
+      c = code(w₀*σ)
+      return([n-c[i] for i in 1:n])
+   end
+
+   n = length(lambda)
+   N = binomial(n+1,2)
+   b = flag(w)
+   P =  Matrix{Int64}(zeros(n,n))#this will set a bijection between p[i,j] (as in Postnikov/Stanley '09 and the i-th coordinate of GT's ambient space)
+   varctr = N #starting at the last coordinate
+   matctr = [1,1]
+   while varctr>0
+      P[matctr...]=varctr
+      if matctr[2]==1
+         matctr[1]=matctr[1]+1
+         matctr[2]=matctr[1]
+      else
+      	 matctr[2]=matctr[2]-1
+      end
+      varctr=varctr-1
+   end
+   Hyperplanes = []
+   for i in 1:n #for each i there's a set of variables p_(i,j) which need to be equal
+      startVarIndex = P[n,i]
+      for j in b[i]:n-1
+         varIndex = P[j,i]
+         H = []
+         for k in 1:N
+            if k==startVarIndex
+               push!(H,1)
+            elseif k==varIndex
+               push!(H,-1)
+            else
+               push!(H,0)
+            end
+         end
+         push!(Hyperplanes,H)
+      end
+   end
+   if length(Hyperplanes)==0
+      return(gelfand_tsetlin_polytope(lambda))
+   end
+   z = [0 for i in 1:length(Hyperplanes)]
+   A = transpose(hcat(Hyperplanes...))
+   #now L is the intersection of the linear spaces Ax = 0 = z
+   L = Polyhedron(([0 for i in 1:N],[0]),(A,z))
```

Comment:
```suggestion
   L = polyhedron(nothing,(A,z))
```
`nothing` can be used instead of an empty inequality, and the uppercase `Polyhedron` constructor has been deprecated in favor of the lowercase function call.